ols/app/endpoints/ols.py

```diff
@@ -31,16 +31,14 @@ def conversation_request(llm_request: LLMRequest) -> LLMResponse:
     """
     # Initialize variables
     previous_input = None
-    conversation_id = llm_request.conversation_id
 
     # TODO: retrieve proper user ID from request
     user_id = "user1"
 
-    # Generate a new conversation ID if not provided
-    if not conversation_id:
-        conversation_id = suid.get_suid()
-        logger.info(f"{conversation_id} New conversation")
-    else:
+    conversation_id = retrieve_conversation_id(llm_request)
+
+    # TODO: will be refactored in following PR
+    if llm_request.conversation_id:
```

Comment:
I did a PR a while ago when I was adding the generation of uuid in the pydantic model itself (in case it is missing). But in that case, some other attr needs to be set to know if the id was provided or generated.

This is just FYI. It is probably better to not handle it model.